compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3716,7 +3716,7 @@ private BLangExpression createPatternIfCondition(BLangMatchBindingPatternClause
         BType[] memberTypes;
         if (patternType.tag == TypeTags.UNION) {
             BUnionType unionType = (BUnionType) patternType;
-            memberTypes = unionType.memberTypes.toArray(new BType[0]);
+            memberTypes = unionType.getMemberTypes().toArray(new BType[0]);
```

Comment:
@pubudu91 Shall we return a Collections.unmodifiableList(list), to prevent adding into list my mistake?

Reply:
+1

Reply:
Addressed with 46f4db7